modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/application/ApplicationDescriptorFactoryTestCase.java

```diff
@@ -84,11 +84,16 @@ private static File getResourceFile(String resource) {
   public static final String JAR_FILE_NAME = "test.jar";
 
   @Rule
-  public SystemProperty remoteMavenRepository = new SystemProperty("muleRuntimeConfig.maven.repositoryLocation",
-                                                                   discoverProvider(ApplicationDescriptorFactoryTestCase.class
-                                                                       .getClassLoader()).getLocalRepositorySuppliers()
-                                                                           .environmentMavenRepositorySupplier().get()
-                                                                           .getAbsolutePath());
+  public SystemProperty repositoryLocation = new SystemProperty("muleRuntimeConfig.maven.repositoryLocation",
+                                                                discoverProvider(ApplicationDescriptorFactoryTestCase.class
+                                                                    .getClassLoader()).getLocalRepositorySuppliers()
+                                                                        .environmentMavenRepositorySupplier().get()
+                                                                        .getAbsolutePath());
+
+  @Rule
+  public SystemProperty remoteMavenRepository =
```

Comment:
Can't you use M2_REPO instead or the remote one? If it is required to go remote to find an artifact, then that is wrong to me, every artifact required on the test should be already built or downloaded as part of another module build.